commit e6c02fc40a01f43e15b44e279c75757dd2b15b65 (HEAD -> master)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2019-09-16T16:27:58+02:00 (3 weeks ago)
    
    TRITON-1894 svc:/smartdc/agent/firewaller-config-migration:default goes into maint
    Reviewed by: Trent Mick <trentm@gmail.com>
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/config-migration.js b/config-migration.js
index c1f5644..7c34093 100644
--- a/config-migration.js
+++ b/config-migration.js
@@ -77,7 +77,17 @@ vasync.pipeline({
                 func: function testIPFFiles(vm, nextVm) {
                     var ipfv4 = util.format(IPF_CONF,
                         ctx.vmsByVmUUID[vm.uuid].zonepath);
-                    var ipfv4Data = fs.readFileSync(ipfv4, 'utf8');
+                    var ipfv4Data;
+                    try {
+                        ipfv4Data = fs.readFileSync(ipfv4, 'utf8');
+                    } catch (err) {
+                        if (err.code !== 'ENOENT') {
+                            nextVm(err);
+                            return;
+                        }
+                        nextVm();
+                        return;
+                    }
                     var ipfv4Res = IPF_VER_RE.exec(ipfv4Data);
                     // If we cannot find a version written on the rules file,
                     // let's assume it's version 1 (pre RFD 163):
